internal/api/courses/admin/email.go

```diff
@@ -0,0 +1,76 @@
+package admin
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/email"
+)
+
+type EmailRequest struct {
+	core.APIRequestCourseUserContext
+	core.MinCourseRoleGrader
+
+	email.Message
+
+	DryRun bool `json:"dry-run"`
+}
+
+type EmailResponse struct {
+	To  []string `json:"to"`
+	CC  []string `json:"cc"`
+	BCC []string `json:"bcc"`
+}
+
+// Send an email to course users.
+func HandleEmail(request *EmailRequest) (*EmailResponse, *core.APIError) {
+	response := EmailResponse{}
+	var err error
+
+	if request.Subject == "" {
+		return nil, core.NewBadRequestError("-627", &request.APIRequest, "No email subject provided.")
+	}
+
+	var errs error
+
+	request.To, err = db.ResolveCourseUsers(request.Course, request.To)
+	if err != nil {
+		err = fmt.Errorf("Failed to resolve 'to' email addresses.")
+		errs = errors.Join(errs, err)
+	}
```

Comment:
You missed the point of an errors.Join() a bit.
You don't need to make a new error or have a conditional when using it.

If you really want the specific error message you can use the conditionals,
but then specific internal errors would make more sense.